Lab2.ipynb

```diff
@@ -0,0 +1,677 @@
+    "x2 = np.linspace(0, 10, len(conv2))\n",
+    "x3 = np.linspace(0, 10, len(conv3))\n",
+    "x4 = np.linspace(0, 10, len(conv4))\n",
+    "\n",
+    "fig, ax = plt.subplots(1, 1, figsize=(15, 5))\n",
+    "\n",
+    "plt.step(x1, conv1, label = 'First Convolution')\n",
+    "plt.step(x2, conv2, label = 'Second Convolution')\n",
+    "plt.step(x3, conv3, label = 'Third Convolution')\n",
+    "plt.step(x4, conv4, label = 'Fourth Convolution')\n",
+    "plt.title('Poisson PMF of Cosmic Rays Averaged')\n",
+    "plt.xlabel(\"Average Observed Cosmic Rays\")\n",
+    "plt.ylabel('Probability Mass')\n",
+    "plt.legend(loc='best')\n",
+    "plt.show()\n",
+    "\n",
+    "# semilog graph\n",
+    "plt.step(x1, conv1, label = 'First Convolution')\n",
+    "plt.step(x2, conv2, label = 'Second Convolution')\n",
+    "plt.step(x3, conv3, label = 'Third Convolution')\n",
+    "plt.step(x4, conv4, label = 'Fourth Convolution')\n",
+    "plt.title('Log-Poisson PMF of Comsic Rays Averaged')\n",
+    "plt.xlabel(\"Average Observed Cosmic Rays\")\n",
+    "plt.ylabel('Log of Probability Mass')\n",
+    "plt.legend(loc='best')\n",
+    "plt.semilogy()\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "841d3e2c-f702-4c9c-bf99-2b8c3b05cf5d",
+   "metadata": {},
+   "source": [
+    "1C: As we average the days (increase number of convolve functions), the probability distribution retains its shape but becomes skinnier, while still centered around the same mean. In relation to part B, since the summed probability distribution is still Poisson, the distribution when we average days should also remain Poisson. Looking at the semilog graph, it is apparant that as the number of days increases, the shape of the distribution becomes more and more like an upside down parabola, which is in line with what the Central Limit Theorem states. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "9f398718-2d01-4e87-88ad-267afb44008e",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "1.4082807867874816\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Part 1D\n",
+    "conv5 = np.convolve(conv4, dist)\n",
+    "events = (6 * 8) # chose N = 6, Y = 8\n",
+    "cdf = 0\n",
+    "for i in range(int(events + 1)):\n",
+    "    cdf += conv5[i]\n",
+    "print(np.abs(stats.norm.ppf(1-cdf)))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "14b72f0a-8e49-4090-87c6-4efa416a0c07",
+   "metadata": {},
+   "source": [
+    "1D: Choosing N = 6 days and Y = 8  average gamma rays emitted, gives the total number of events as 6 * 8 = 48 events observed. Then we must integrate from 0 to 48 to calculate the probability that the background noise gave a signal less than the detected value. We must then subtract this probability from 1 to get the probability that the background noise gave a signal larger than what was expected. Sigma = 1.41\n",
```

Comment:
Also in 1D, we consider large values more signal like, so we want to integrate from 48 events to infinity, not 0 to 48 events. You can do this by iterating over `range(events, len(conv4))` instead of `range(0, events)`